crates/ruff_linter/src/rules/refurb/rules/read_whole_file.rs

```diff
@@ -129,21 +129,43 @@ impl<'a> Visitor<'a> for ReadMatcher<'a, '_> {
                     return;
                 }
 
-                let target = match self.with_stmt.body.first() {
+                let (target, annotation) = match self.with_stmt.body.first() {
                     Some(Stmt::Assign(assign))
                         if assign.value.range().contains_range(expr.range()) =>
                     {
                         match assign.targets.first() {
-                            Some(Expr::Name(name)) => Some(name.id.as_str()),
-                            _ => None,
+                            Some(Expr::Name(name)) => (Some(name.id.as_str()), None),
+                            _ => (None, None),
                         }
                     }
-                    _ => None,
+                    Some(Stmt::AnnAssign(ann_assign))
+                        if ann_assign
+                            .value
+                            .as_ref()
+                            .is_some_and(|v| v.range().contains_range(expr.range())) =>
+                    {
+                        match ann_assign.target.as_ref() {
+                            Expr::Name(name) => {
+                                let annotation_code = self
+                                    .checker
+                                    .generator()
+                                    .expr(ann_assign.annotation.as_ref());
```

Comment:
Do we need to use the generator here, or can we just slice the `&str` out of the input?


```suggestion
                                let annotation_code = self
                                    .checker
                                    .locator()
                                    .slice(ann_assign.annotation.range());
```

(untested but that's what I had in mind)